services/worker/src/worker/job_runners/split/opt_in_out_urls_scan_from_streaming.py

```diff
@@ -224,7 +224,7 @@ def compute_opt_in_out_urls_scan_response(
 
     # get the urls
     num_scanned_rows = len(rows)
-    urls = [row[urls_column] for row in rows for urls_column in image_url_columns]
+    urls = [row[urls_column] if row[urls_column] else "" for row in rows for urls_column in image_url_columns]
```

Comment:
shouldn't we filter instead?

```suggestion
    urls = [row[urls_column] for row in rows for urls_column in image_url_columns if row[urls_column]]
```

Reply:
I thought the same, but it will affect when matching the opt indices at https://github.com/huggingface/datasets-server/blob/main/services/worker/src/worker/job_runners/split/opt_in_out_urls_scan_from_streaming.py#L52

Reply:
The problem is when sending the concatenated list of URLs at  https://github.com/huggingface/datasets-server/blob/main/services/worker/src/worker/job_runners/split/opt_in_out_urls_scan_from_streaming.py#L38 we have to keep the order or maybe think another logic to map the URL to the index.